plugins/catalog-backend-module-bitbucket-cloud/src/service/BitbucketCloudCatalogModule.test.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ConfigReader } from '@backstage/config';
+import { getVoidLogger } from '@backstage/backend-common';
+import {
+  configServiceRef,
+  loggerServiceRef,
+  schedulerServiceRef,
+} from '@backstage/backend-plugin-api';
+import {
+  PluginTaskScheduler,
+  TaskScheduleDefinition,
+} from '@backstage/backend-tasks';
+import { startTestBackend } from '@backstage/backend-test-utils';
+import { catalogProcessingExtensionPoint } from '@backstage/plugin-catalog-node';
+import { bitbucketCloudCatalogModule } from './BitbucketCloudCatalogModule';
+import { Duration } from 'luxon';
+import { BitbucketCloudEntityProvider } from '../BitbucketCloudEntityProvider';
+
+describe('bitbucketCloudCatalogModule', () => {
+  it('should register provider at the catalog extension point', async () => {
+    let addedProviders: Array<BitbucketCloudEntityProvider> | undefined;
+    let usedSchedule: TaskScheduleDefinition | undefined;
+
+    const extensionPoint = {
+      addEntityProvider: (providers: any) => {
+        addedProviders = providers;
+      },
+    };
+    const runner = jest.fn();
+    const scheduler = {
+      createScheduledTaskRunner: (schedule: TaskScheduleDefinition) => {
+        usedSchedule = schedule;
+        return runner;
+      },
+    } as unknown as PluginTaskScheduler;
+
+    const config = new ConfigReader({
+      catalog: {
+        providers: {
+          bitbucketCloud: {
+            schedule: {
+              frequency: 'P1M',
+              timeout: 'PT3M',
+            },
+            workspace: 'test-ws',
+          },
+        },
+      },
+    });
+
+    await startTestBackend({
```

Comment:
Just pointing out the obvious flaw that there is no way to stop this, we're we'll be fixing soon 😅